homeassistant/components/efesto/climate.py

```diff
@@ -0,0 +1,218 @@
+            ATTR_SMOKE_TEMP: self._smoke_temperature,
+            ATTR_REAL_POWER: self._real_power,
+            ATTR_OPERATION_MODE: self._operation_mode,
+        }
+        return data
+
+    @property
+    def name(self):
+        """Return the name of the Efesto, if any."""
+        return self._name
+
+    @property
+    def precision(self):
+        """Return the precision of the system."""
+        return PRECISION_WHOLE
+
+    @property
+    def target_temperature_step(self):
+        """Return the supported step of target temperature."""
+        return PRECISION_WHOLE
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return TEMP_CELSIUS
+
+    @property
+    def is_on(self):
+        """Return true if on."""
+        return self._on
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._target_temperature
+
+    def turn_off(self):
+        """Turn device off."""
+        try:
+            json = self.client.set_off()
+            _LOGGER.debug("JSON OUTPUT: %s", json)
+            if json['status'] > 0:
+                _LOGGER.error("Failed to turn off device (%s)",
+                              json['message'])
+                return
+
+        except ValueError:
+            _LOGGER.error("Failed to turn off device")
+            return
+
+    def turn_on(self):
+        """Turn device on."""
+        try:
+            json = self.client.set_on()
+            _LOGGER.debug("JSON OUTPUT: %s", json)
+            if json['status'] > 0:
+                _LOGGER.error("Failed to turn on device (%s)", json['message'])
+                return
+
+        except ValueError:
+            _LOGGER.error("Failed to turn on device")
+            return
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        if temperature is None:
+            return
+        res = self.client.set_temperature(round(temperature))
+        if res['status'] > 0:
+            _LOGGER.error("Failed to set temperature")
+            return
+
+    def update(self):
+        """Get the latest data."""
+        try:
+            json = self.client.get_status()
+            _LOGGER.debug("JSON OUTPUT: %s", json)
+
+            if json['status'] == 0:
+                self._device_status = json['deviceStatus']
+                if self._device_status == 0:
+                    self._on = False
+                else:
+                    self._on = True
+                    if 'idle_info' in json:
+                        self._idle_info = json['idle_info']
+                        if self._idle_info == "TURNING OFF":
+                            self._on = False
+
+                if json['airTemperature']:
```

Comment:
This does not check if key is missing. Are they always there but "none" orare they missing completely?

Reply:
They should be always there but can be None.
So better to update to `if json['airTemperature'] is not None:` ?

Reply:
No then current code is fine. If it had been the opposite using .get(attrname, default) would have been better

Reply:
Thanks for the tip!